### PR TITLE
[pylint] pylint and flake8 updates

### PR DESCRIPTION
**Pull request by arif-ali**
pylint for the following rules applied

* C0411: wrong-import-order
* R0912: too-many-branches
* R0914: too-many-locals
* R1725: super-with-arguments
* E1101: no-member

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [ ] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: sos/collector/sosnode.py [hunk 1]
```diff
@@ -838,36 +840,25 @@ def remove_file(self, path):
...
-        if self.sos_path:
-            if self.need_sudo or self.opts.become_root:
-                try:
-                    self.make_archive_readable(self.sos_path)
-                except Exception:
-                    self.log_error('Failed to make archive readable')
-                    return False
-            self.log_info(f'Retrieving sos report from {self.address}')
-            self.ui_msg('Retrieving sos report...')
+        if self.need_sudo or self.opts.become_root:
```
Comment (arif-ali) [inline sos/collector/sosnode.py:843]: The reason why this was refactored was, where `retrieve_sosreport()` function is being used it already checks for `self.sos_path`, so the code under the else seemed redundant

**Comment by arif-ali** [comment]
`pylint` now looks great with the 2 PRs for f-strings done. Based on the output of `pylint` here, #3604 should resolve the rest.

This way, we can then enable it, so that future PRs we'll get errors for f-strings too.

File: sos/collector/transports/__init__.py [hunk 2]
```diff
@@ -322,7 +323,7 @@ def _send_pexpect_password(self, index, result):
         :type result:       ``pexpect.spawn``
         """
         if index == 2:
-            if not self.opts.sudo_pw and not self.opt.nopasswd_sudo:
+            if not self.opts.sudo_pw and not self.opts.nopasswd_sudo:
```
Comment (pmoravec) [inline sos/collector/transports/__init__.py:326]: Nice catch.

File: sos/policies/distros/redhat.py [hunk 3]
```diff
@@ -624,7 +623,8 @@ def check(cls, remote=''):
 
     def fedora_version(self):
         pkg = self.pkg_by_name("fedora-release") or \
-            self.all_pkgs_by_name_regex("fedora-release-.*")[-1]
+            self.package_manager.all_pkgs_by_name_regex(
+                "fedora-release-.*")[-1]
```
Comment (pmoravec) [inline sos/policies/distros/redhat.py:627]: How this could even work now..? good catch.

**Review by pmoravec — APPROVED** [review]

File: sos/collector/transports/__init__.py (hunk 2 above)
Comment (arif-ali) [reply sos/collector/transports/__init__.py:326]: I didn't catch it, `pylint` did

File: sos/policies/distros/redhat.py (hunk 3 above)
Comment (arif-ali) [reply sos/policies/distros/redhat.py:627]: I didn't catch it, `pylint` did

**Pull request merged**